modules/php/Actions/Scenario2/StirWaterTanks.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Bga\Games\WelcomeToTheMoon\Actions\Scenario2;
+
+use Bga\Games\WelcomeToTheMoon\Core\Notifications;
+use Bga\Games\WelcomeToTheMoon\Models\Player;
+
+class StirWaterTanks extends \Bga\Games\WelcomeToTheMoon\Models\Action
+{
+  public function getState(): int
+  {
+    return ST_STIR_WATER_TANKS;
+  }
+
+  public function isOptional(): bool
+  {
+    return true;
+  }
+
+  public function isDoable(Player $player): bool
+  {
+    return !empty($this->getArgs()['slots']);
+  }
+
+  public function argsStirWaterTanks()
+  {
+    $slot = $this->getCtxArgs()['slot'];
+    $isWaterTankAttached = in_array($slot, array_keys(self::$waterTanksAtSlots));
```

Comment:
I suggest you to use the null coalescing operator instead :
$slots = self::$waterTanksAtSlots[slot] ?? []